flask_jwt_extended/internal_utils.py

```diff
@@ -25,11 +25,11 @@ def user_lookup(*args, **kwargs):
     return jwt_manager._user_lookup_callback(*args, **kwargs)
 
 
-def verify_token_type(decoded_token, refresh):
-    if not refresh and decoded_token["type"] == "refresh":
-        raise WrongTokenError("Only non-refresh tokens are allowed")
-    elif refresh and decoded_token["type"] != "refresh":
+def verify_token_type(decoded_token, refresh, token_type):
+    if refresh and decoded_token["type"] != "refresh":
         raise WrongTokenError("Only refresh tokens are allowed")
+    elif not refresh and decoded_token["type"] != token_type:
+        raise WrongTokenError(f"Token of type { decoded_token['type'] } is not allowed")
```

Comment:
I think this might be a breaking change. I believe some people are using JWTs from other sources that set the `type` to other fields, and we implictly treat those as access tokens right now. 

I haven't fully thought this out, but I wonder if we instead set `token_type` to `null` in the view decorators and `verify_jwt_in_request`, and then we preserve the old logic here unless a `token_type` is actually specified?